source/postprocess/topography_to_file.cc

```diff
@@ -0,0 +1,290 @@
+#include <deal.II/fe/fe_values.h>
+
+#include <cmath>
+#include <limits>
+
+
+
+namespace aspect
+{
+  namespace Postprocess
+  {
+    template <int dim>
+    std::pair<std::string,std::string>
+    TopographyToFile<dim>::execute (TableHandler &statistics)
+    {
+      const types::boundary_id relevant_boundary = this->get_geometry_model().translate_symbolic_boundary_name_to_id ("top");
+
+      // Get a quadrature rule that exists only on the corners
+      QTrapez<dim-1> face_corners;
+      FEFaceValues<dim> face_vals (this->get_mapping(), this->get_fe(), face_corners, update_quadrature_points);
+
+      // have a stream into which we write the data. the text stream is then
+      // later sent to processor 0
+      std::ostringstream output_stats;
+      std::ostringstream output_file;
+      std::vector<std::pair<Point<dim>,double> > stored_values;
+
+
+      // Choose stupidly large values for initialization
+      double local_max_height = -std::numeric_limits<double>::max();
+      double local_min_height = std::numeric_limits<double>::max();
+
+      // loop over all of the surface cells and save the elevation to stored_value
+      typename parallel::distributed::Triangulation<dim>::active_cell_iterator cell = this->get_triangulation().begin_active(),
+                                                                               endc = this->get_triangulation().end();
+
+      for (; cell != endc; ++cell)
+        if (cell->is_locally_owned() && cell->at_boundary())
+          for (unsigned int face_no = 0; face_no < GeometryInfo<dim>::faces_per_cell; ++face_no)
+            if (cell->face(face_no)->at_boundary())
+              {
+                if ( cell->face(face_no)->boundary_id() != relevant_boundary)
+                  continue;
+
+                face_vals.reinit( cell, face_no);
+
+                for (unsigned int corner = 0; corner < face_corners.size(); ++corner)
+                  {
+                    Point<dim> vertex = face_vals.quadrature_point(corner);
+                    double elevation = this->get_geometry_model().height_above_original_surface(vertex);
+                    stored_values.push_back (std::make_pair(vertex, elevation));
+                    if ( elevation > local_max_height)
+                      local_max_height = elevation;
+                    if ( elevation < local_min_height)
+                      local_min_height = elevation;
+                  }
+              }
+
+      //Calculate min/max topography across all processes
+      const double max_topography = Utilities::MPI::max(local_max_height, this->get_mpi_communicator());
+      const double min_topography = Utilities::MPI::min(local_min_height, this->get_mpi_communicator());
+
+      //Write results to statistics file
+      statistics.add_value ("Minimum topography (m)",
+                            min_topography);
+      statistics.add_value ("Maximum topography (m)",
+                            max_topography);
+      //{
```

Comment:
remove this if it isn't used. Same for the \\} below

Reply:
Sure. I copied this part from topography.cc and wasn't sure if it was just me or the curly braces were not needed.